### ISSUE TITLE
Improve tab-completion

### ISSUE THREAD

**Issue by dhirschfeld** [nteract/archived-desktop-app#1322]
The current tab-completion seems to be somewhat limited - specifically in that it can't handle completion of nested objects. A further bugbear is that it won't trigger completion on a period but requires you to have finished typing the object name immediately before typing the period.
![completion](https://cloud.githubusercontent.com/assets/881019/21832952/85085c38-d7fa-11e6-85f0-3b6ecfb20f38.gif)

**Comment by dhirschfeld** [comment]
For reference, here's the same from my IPython (Spyder) session:
![completion2](https://cloud.githubusercontent.com/assets/881019/21832999/c7a41be0-d7fa-11e6-98c7-c1ac5ae6a51a.gif)

**Comment by rgbkrk** [comment]
Wow I'm surprised that the embedded terminal in Spyder has such great support for nested objects! Thanks for reporting this.

My jupyter notebook server and local IPython both show this:

<img width="443" alt="screen shot 2017-01-10 at 6 43 17 pm" src="https://cloud.githubusercontent.com/assets/836375/21833083/acc47e0a-d764-11e6-8104-b37161f71121.png">

**Comment by dhirschfeld** [comment]
Ah, that'll be because I use the greedy completer by default - it's *very* useful when mucking about interactively - which I think nteract is great for! You can test in IPython with:
 `%config IPCompleter.greedy=True`

You can't seem to enable that in nteract though? Also there remains the issue of having to delete the last letter of the object before typing the period to trigger completion. It seems tab doesn't trigger completion - is there any way to force it (other than deleting and retyping the last letter and period)?

**Comment by rgbkrk** [comment]
Tab does not trigger completion in nteract. There is a way to trigger the completion though: ctrl-space. Since tab is also setup for indent, the [logic for tab-completion in Jupyter is a bit hairy](https://github.com/jupyter/notebook/blob/179bb24fbf79d153812858126127a91431da3319/notebook/static/notebook/js/completer.js). We simplified the approach for now to have something while lowering our maintenance burden.

It looks like we support the greedy completions -- our typeahead is apparently holding back some completions that I had to trigger with ctrl-space:

![screen shot 2017-01-10 at 7 48 35 pm](https://cloud.githubusercontent.com/assets/836375/21834933/cc49c97a-d76d-11e6-8a04-3c9f5533fdfe.png)
